File: leetcode-oj-impl/src/cn/com/pingouin/leet_code/oj/impl/TwoSum.java
package cn.com.pingouin.leet_code.oj.impl;

import java.util.Arrays;
import java.util.HashMap;

public class TwoSum {

	public static int[] twoSum(int[] numbers, int target) {
        
        int[] result = new int[2];

		HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();

        for(int i=0, len=numbers.length; i<len; i++) {
            if (map.containsKey(numbers[i])) {
            	result[0] = i+1;
            	result[1] = map.get(numbers[i])+1;
                
                break;
            }
            else map.put(target-numbers[i], i);
        }

        Arrays.sort(result);
		
		return result;
    }
	
	public static void main(String[] args) {
		int[] numbers = {5, 25,75};
		int target = 100;
		
		int[] result = twoSum(numbers, target);
		
		System.out.println(result[0] + " @@@ " + result[1]);
	}
	
}

File: leetcode-oj-impl/src/cn/com/pingouin/leet_code/oj/js/TwoSum.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]} two integers in an array, ie: [index1, index2]
 */

/**
 * 和java解法一样
 * 数组排序时需要自定义一个函数作为排序依据
 */
 
function sortNum(a, b) {
    return a - b;
}

var twoSum = function(numbers, target) {
    var result = [];
    
    var map = {};
    
    for(i=0; i<numbers.length; i++) {
        if(map.hasOwnProperty(numbers[i].toString())) {
            result[0] = i + 1;
            result[1] = map[numbers[i].toString()] + 1;
            
            break;
        }
        else {
            var dValue = target - numbers[i];
            map[dValue.toString()] = i;
        }
    }
    
    return result.sort(sortNum);
};